tools/ec2info/main.go

```diff
@@ -59,3 +57,167 @@ func main() {
 	check(write(f, flat, pkg))
 	check(format(output))
 }
+
+func environment() error {
+	if value := os.Getenv("AWS_ACCESS_KEY_ID"); value == "" {
+		return errors.New("AWS_ACCESS_KEY_ID must be set")
+	}
+	if value := os.Getenv("AWS_SECRET_ACCESS_KEY"); value == "" {
+		return errors.New("AWS_SECRET_ACCESS_KEY must be set")
+	}
+	if value := os.Getenv("AWS_SESSION_TOKEN"); value == "" {
+		return errors.New("AWS_SESSION_TOKEN must be set")
+	}
```

Comment:
If you're using doormat to set your credentials these values won't be set in the environment, you'll instead have short-lived credentials in `~/.aws/credentials`. A better way to do this might be to capture the error from `NewSession` and make a suggestion that the AWS credentials need to be setup (either via env var or doormat)

Reply:
Good catch, fixed!

```
➜ make ec2info 
==> Generating AWS EC2 specifications ...
2022/09/09 08:47:57 failed to create AWS session; make sure environment is setup
2022/09/09 08:47:57 must have environment variables AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN
2022/09/09 08:47:57 or ~/.aws/credentials configured properly
2022/09/09 08:47:57 NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
exit status 1
```